### PR TITLE
Update Kubernetes to v1.26.14

### PR DESCRIPTION
**Pull request by brooksn**
Updates Kubernetes to v1.26.14 and Go to 1.21.7.

### CODE REVIEW

**Review by tashima42 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/k3s-io/k3s
 
-go 1.19
+go 1.21
```
Comment (brandond) [inline go.mod:3]: Upstream is still on `go 1.19` in go.mod, can we leave this as-is?
https://github.com/kubernetes/kubernetes/blob/release-1.26/go.mod#L9

```suggestion
go 1.19
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (brooksn) [reply go.mod:3]: Upstream is building with go 1.21, and running go mod tidy updates this version as follows anyway:
```
go 1.21

toolchain go1.21.7
```
Setting GOTOOLCHAIN doesn't work for me, but anything other than running go mod tidy with an old go version results in the go mod file being updated.

Comment (brandond) [reply go.mod:3]: yeah sounds like that's unavoidable :/

**Review by galal-hussein — APPROVED** [review]

**Comment by brandond** [comment]
can we squash this to remove the various change/revert-the-change commits that are in here? Or at least drop the change/revert pairs from this branch prior to merging?

**Review by brandond — APPROVED** [review]

**Review by tashima42 — APPROVED** [review]

**Pull request merged**